node/src/components/network/tests_bulk_gossip.rs

```diff
@@ -0,0 +1,240 @@
+      collector = Event<DummyPayload>;
+  }
+
+  requests: {
+      NetworkRequest<NodeId, DummyPayload> -> net;
+  }
+
+  announcements: {
+      NetworkAnnouncement<NodeId, DummyPayload> -> [collector];
+  }
+});
+
+impl NetworkedReactor for LoadTestingReactor {
+    type NodeId = NodeId;
+
+    fn node_id(&self) -> Self::NodeId {
+        self.net.node_id()
+    }
+}
+
+/// Configuration for the test reactor.
+#[derive(Debug)]
+pub struct TestReactorConfig {
+    /// The fixed chainspec used in testing.
+    chainspec: Chainspec,
+    /// Network configuration used in testing.
+    network_config: crate::components::network::Config,
+}
+
+/// A dummy payload.
+#[derive(Clone, Eq, Deserialize, PartialEq, Serialize)]
+pub struct DummyPayload(Vec<u8>);
+
+const DUMMY_PAYLOAD_ID_LEN: usize = 16;
+
+/// ID of a dummy payload.
+type DummyPayloadId = [u8; DUMMY_PAYLOAD_ID_LEN];
+
+impl DummyPayload {
+    /// Creates a new randomly generated payload.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `sz` is less than `DUMMY_PAYLOAD_ID_LEN` bytes.
+    fn random_with_size(rng: &mut TestRng, sz: usize) -> Self {
+        assert!(
+            sz >= DUMMY_PAYLOAD_ID_LEN,
+            "payload must be large enough to derive ID"
+        );
+
+        DummyPayload(rng.sample_iter(Standard).take(sz).collect())
+    }
+
+    /// Returns the ID of the payload.
+    fn id(&self) -> DummyPayloadId {
+        TryFrom::try_from(&self.0[..DUMMY_PAYLOAD_ID_LEN])
+            .expect("could not get ID data from buffer slice")
+    }
+}
+
+impl Collectable for DummyPayload {
+    type CollectedType = DummyPayloadId;
+
+    fn into_collectable(self) -> Self::CollectedType {
+        self.id()
+    }
+}
+
+impl Debug for DummyPayload {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "payload ({} bytes: {:?}...)",
+            self.0.len(),
+            &self.0[0..self.0.len().min(10)]
+        )
+    }
+}
+
+impl Display for DummyPayload {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        Debug::fmt(self, f)
+    }
+}
+
+#[tokio::test]
+async fn send_large_message_across_network() {
+    init_logging();
+
+    if env::var(ENABLE_LIBP2P_ENV_VAR).is_err() {
+        eprintln!("{} not set, skipping test", ENABLE_LIBP2P_ENV_VAR);
+        return;
+    }
+
+    // This can, on a decent machine, be set to 30, 50, maybe even 100 nodes. The default is set to
+    // 5 to avoid overloading CI.
+    let node_count: usize = 5;
+
+    // Fully connecting a 20 node network takes ~ 3 seconds. This should be ample time for gossip
+    // and connecting.
+    let timeout = Duration::from_secs(60);
+    let large_size: usize = 1024 * 1024 * 4;
+
+    let mut rng = crate::new_rng();
+
+    // Port for first node, other will connect to it.
+    let first_node_port = testing::unused_port_on_localhost() + 1;
+
+    let mut net = testing::network::Network::<LoadTestingReactor>::new();
```

Comment:
I understand that you highly disagree, but we've already discussed this and we have a style guide to avoid wasting time on issues like this.  Please amend as per the coding standards and the rest of the codebase.

Reply:
"Fixed".